code/modules/antagonists/traitor/IAA/internal_affairs.dm

```diff
@@ -1,247 +1,209 @@
+	return ..()
 
-/datum/antagonist/traitor/internal_affairs/apply_innate_effects()
+/datum/antagonist/traitor/internal_affairs/apply_innate_effects(mob/living/mob_override)
 	. = ..()
+	var/mob/living/datum_owner = mob_override || owner.current
+	datum_owner.apply_status_effect(/datum/status_effect/agent_pinpointer)
 
-	if(!owner)
-		CRASH("Antag datum with no owner.")
+/datum/antagonist/traitor/internal_affairs/remove_innate_effects(mob/living/mob_override)
+	var/mob/living/datum_owner = mob_override || owner.current
+	datum_owner.remove_status_effect(/datum/status_effect/agent_pinpointer)
+	return ..()
 
-	if(owner.current)
-		give_pinpointer(owner.current)
+/// Affairs Agents can only roll Internal or External, rather than the normal factions.
+/datum/antagonist/traitor/internal_affairs/pick_employer()
+	var/faction = prob(EXTERNAL_CHANCE) ? ROLE_EXTERNAL_AFFAIRS : ROLE_INTERNAL_AFFAIRS
 
-/datum/antagonist/traitor/internal_affairs/remove_innate_effects()
-	. = ..()
+	employer = faction
 
-	if(!owner)
-		CRASH("Antag datum with no owner.")
+	traitor_flavor = strings(IAA_FLAVOR_FILE, employer)
 
-	if(owner.current)
-		owner.current.remove_status_effect(/datum/status_effect/agent_pinpointer)
+	// External Affairs get an additional objective, done here since objectives are already assigned
+	if(employer == ROLE_EXTERNAL_AFFAIRS)
+		owner.special_role = ROLE_EXTERNAL_AFFAIRS
+		should_give_codewords = TRUE
+		objectives += forge_single_generic_objective()
 
-/datum/antagonist/traitor/internal_affairs/on_gain()
-	START_PROCESSING(SSprocessing, src)
-	. = ..()
-/datum/antagonist/traitor/internal_affairs/on_removal()
-	STOP_PROCESSING(SSprocessing,src)
+/datum/antagonist/traitor/internal_affairs/greet()
 	. = ..()
-/datum/antagonist/traitor/internal_affairs/process()
-	iaa_process()
-
-/datum/status_effect/agent_pinpointer
-	id = "agent_pinpointer"
-	duration = -1
-	tick_interval = PINPOINTER_PING_TIME
-	alert_type = /atom/movable/screen/alert/status_effect/agent_pinpointer
-	var/minimum_range = PINPOINTER_MINIMUM_RANGE
-	var/range_fuzz_factor = PINPOINTER_EXTRA_RANDOM_RANGE
-	var/mob/scan_target = null
-	var/range_mid = 8
-	var/range_far = 16
-
-/atom/movable/screen/alert/status_effect/agent_pinpointer
-	name = "Internal Affairs Integrated Pinpointer"
-	desc = "Even stealthier than a normal implant."
-	icon = 'icons/obj/device.dmi'
-	icon_state = "pinon"
-
-/datum/status_effect/agent_pinpointer/proc/point_to_target() //If we found what we're looking for, show the distance and direction
-	if(!scan_target)
-		linked_alert.icon_state = "pinonnull"
-		return
-	var/turf/here = get_turf(owner)
-	var/turf/there = get_turf(scan_target)
-	if(here.z != there.z)
-		linked_alert.icon_state = "pinonnull"
-		return
-	if(get_dist_euclidian(here,there)<=minimum_range + rand(0, range_fuzz_factor))
-		linked_alert.icon_state = "pinondirect"
+	crime = pick(
+		"distribution of contraband",
+		"unauthorized erotic action on duty",
```

Comment:
🤨 
how old is this

Reply:
seems to be added in https://github.com/tgstation/tgstation/pull/26651